src/ResourceManager/Network/Commands.Network/VirtualNetworkGatewayConnection/NewAzureRmIpsecPolicyCommand.cs

```diff
@@ -0,0 +1,140 @@
+        [ValidateNotNullOrEmpty]
+        [ValidateSet(
+            MNM.IpsecIntegrity.MD5,
+            MNM.IpsecIntegrity.SHA1,
+            MNM.IpsecIntegrity.SHA256,
+            MNM.IpsecEncryption.GCMAES128,
+            MNM.IpsecEncryption.GCMAES192,
+            MNM.IpsecEncryption.GCMAES256,
+            IgnoreCase = false)]
+        public string IpsecIntegrity { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The IKE encryption algorithm (IKE Phase 2)")]
+        [ValidateNotNullOrEmpty]
+        [ValidateSet(
+            MNM.IkeEncryption.DES,
+            MNM.IkeEncryption.DES3,
+            MNM.IkeEncryption.AES128,
+            MNM.IkeEncryption.AES192,
+            MNM.IkeEncryption.AES256,
+            IgnoreCase = false)]
+        public string IkeEncryption { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The IKE integrity algorithm (IKE Phase 2)")]
+        [ValidateNotNullOrEmpty]
+        [ValidateSet(
+            MNM.IkeIntegrity.MD5,
+            MNM.IkeIntegrity.SHA1,
+            MNM.IkeIntegrity.SHA256,
+            MNM.IkeIntegrity.SHA384,
+            IgnoreCase = false)]
+        public string IkeIntegrity { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The DH Groups used in IKE Phase 1 for initial SA")]
+        [ValidateNotNullOrEmpty]
+        [ValidateSet(
+            MNM.DhGroup.None,
+            MNM.DhGroup.DHGroup1,
+            MNM.DhGroup.DHGroup14,
+            MNM.DhGroup.DHGroup2,
+            MNM.DhGroup.DHGroup2048,
+            MNM.DhGroup.DHGroup24,
+            MNM.DhGroup.ECP256,
+            MNM.DhGroup.ECP384,
+            IgnoreCase = false)]
+        public string DhGroup { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The DH Groups used in IKE Phase 2 for new child SA")]
+        [ValidateNotNullOrEmpty]
+        [ValidateSet(
+            MNM.PfsGroup.None,
+            MNM.PfsGroup.PFS1,
+            MNM.PfsGroup.PFS2,
+            MNM.PfsGroup.PFS2048,
+            MNM.PfsGroup.PFS24,
+            MNM.PfsGroup.ECP256,
+            MNM.PfsGroup.ECP384,
+            IgnoreCase = false)]
+        public string PfsGroup { get; set; } 
+
+        public override void Execute()
+        {
+            base.Execute();
+            var ipsecPolicy = new PSIpsecPolicy();
+
+            ipsecPolicy.SALifeTimeSeconds = this.SALifeTimeSeconds;
+            ipsecPolicy.SADataSizeKilobytes = this.SADataSizeKilobytes;
+            ipsecPolicy.IpsecEncryption = this.IpsecEncryption;
+            ipsecPolicy.IpsecIntegrity = this.IpsecIntegrity;
+            ipsecPolicy.IkeEncryption = this.IkeEncryption;
+            ipsecPolicy.IkeIntegrity = this.IkeIntegrity;
+            ipsecPolicy.DhGroup = this.DhGroup;
+            ipsecPolicy.PfsGroup = this.PfsGroup;
+
+            WriteObject(ipsecPolicy);
```

Comment:
@henry416 as an FYI, StaticAnalysis is going to throw an exception because this cmdlet uses the `New` verb, but doesn't implement `ShouldProcess` . Since this cmdlet is creating a local object, you _do not_ need to implement this attribute, so you can suppress the exception. I will show you how to do so when the time comes 😄